docs/cmd/kn_broker_create.md

```diff
@@ -16,13 +16,26 @@ kn broker create NAME
   # Create a broker 'mybroker' in the 'myproject' namespace and with a broker class of 'Kafka'
   kn broker create mybroker --namespace myproject --class Kafka
 
+  # Create a broker 'mybroker' in the myproject namespace with config referencing configmap named spec-cm
+  kn broker create mybroker --namespace myproject --class Kafka --broker-config cm:spec-cm
+
+  # Create a broker 'mybroker' in the myproject namespace with config referencing configmap named spec-cm in test namespace
+  kn broker create mybroker --namespace myproject --class Kafka --broker-config cm:broker-spec-cm:test
+
+  # Create a broker 'mybroker' in the myproject namespace with config referencing configmap named spec-cm in test namespace
+  kn broker create mybroker --namespace myproject --class Kafka --broker-config cm:broker-spec-cm:test
+
 ```
 
 ### Options
 
 ```
       --backoff-delay string     The delay before retrying.
       --backoff-policy string    The retry backoff policy (linear, exponential).
+      --broker-config string     Reference to the configuration that specifies configuration options for this Broker. For example, a pointer to a ConfigMap, Secret, RabbitmqCluster etc.The format for specifying the object is a colon separated string consisting of at most 3 substrings:
+                                 kind:object-name:namespace=?,apiVersion=?,group=?
+                                 The third substring is optional and the following is also acceptable (in case of ConfigMap, Secret, and RabbitmqCluster kinds):
+                                 kind:object-name
```

Comment:
Proposal for the wording

```
Reference pointing to the broker configuration. The value takes the format "<prefix>:<name>:<namespace>", where <prefix> is either "cm" for a ConfigMap, "secret" for a Secret or a GVK in the format "rabbitmq.com/v1beta1/RabbitmqCluster". "name" is the name of the referenced resource and "namespace" the optional namespace of that resource (which by default is the current namespace).
```

I back @dsimansk opinion that we should not invent another format to provide a full GVK (like in your proposal with additional `apiVersion` and `group`), but encode it already in the prefix. We have done that already I think, for the sink: (or at least plan for this). The prefix actually should fully specify the reference type (and we allow some shortcut for well-known types like cm and secret)

Reply:
Sink is using the same delimiter (':') to get the prefix. So, instead of having the prefix as a part of the first slice and then splitting it by '/', I have added the format as follows:

`<apiVersion>:<kind>:<name>:<namespace>`

I can change this if this doesn't seem right/consistent.